cpp/include/cudf/reshape.hpp

```diff
@@ -16,26 +16,25 @@
 
 #pragma once
 
-#include "cudf.h"
+#include <memory>
+#include <cudf/column/column.hpp>
+#include <cudf/table/table_view.hpp>
 
 namespace cudf {
 
-/**
- * @brief Stack rows of a Table into a single column
- * 
- * Converts the column major table @p in into a row major contiguous buffer,
- * which is returned as a `gdf_column`.
- * Example:
- * ```
- * in = [[4,5,6], [1,2,3]]
- * return = [4,1,5,2,6,3]
- * ```
- * 
- * @note: The dtype of all columns in @p input should be the same
- * 
- * @param input Input table
- * @return gdf_column The result stacked buffer as column
+namespace experimental {
+
+/*
+ * @brief Stacks rows of a table in to a single column.
+ *
+ * Interlaces all columns of a table in to a single column.
```

Comment:
Let's clarify. What does interlaces mean? The old docs are clear, just use them.

Reply:
Using the old docs makes sense, IMO, for the sake of consistency. While attempting to read the documentation, I rewrote it so I could understand what it was actually doing. I thought what I wrote was clear.

Is it just me, or does the word "stack" provide no context as to what this function does, outside of Pandas? "Stack" means to take one thing and put it on top of another. To me, that's the same thing as concatenation, with the added implication of verticality.

If you interlace video or an image, you end up with one row of pixels from each video, in succession, followed by the next row from each video, and so on. That's exactly what this function does (except it does it to the columns in a table, rather than a vector of columns, but we use tables as containers for columns, so the difference seems trivial).

Reply:
We talked about this and you mentioned `interleave_columns` would be a better name. I like it. Should I change the name to `interleave`, or leave it as `stack` for now?

Reply:
Yes, change it. Do you think `interleave` is clear enough? Or `interleave_columns` is better?

Reply:
`interleave_columns` is more specific but also not too verbose, IMHO.

Reply:
I actually think that @devavret is right and this function is a row-major linearize() (2D -> 1D) operation, and more generally `reshape` is probably a better name.

Reply:
@harrism @devavret I'm noticing the reshape/linearize pattern while I'm working on this. I think there's some higher level functions I can extract here, but right now it really is just interleaving the columns. stack + tile are both specific linearization functions, which in turn are specific reshape functions. I see it in my head, I just need to get it in to code. Currently working on making `gather_bitmask_kernel` re-usable for stack / tile. I'll see how far I can get, but the priority is getting `interleave_columns` working (w/nulls and hopefully strings).